### PR TITLE
Become a mentor flow

### PR DESCRIPTION
**Pull request by kntsoriano**
Closes https://github.com/exercism/v3-project-management/issues/138.

### CODE REVIEW

File: app/css/modals/become-mentor.css [hunk 1]
```diff
@@ -81,21 +81,12 @@
...
-                @apply flex flex-col items-center;
-                @apply relative;
-
-                cursor: pointer;
-                transition: background 0.2s ease-out, box-shadow 0.2s ease-out;
-
-                &:hover {
-                    @apply shadow-lgZ1;
-                    @apply bg-backgroundColorA;
+            & .track-checkbox {
```
Comment (kntsoriano) [inline app/css/modals/become-mentor.css]: @iHiD I changed the CSS and HTML a bit in order to make the checkbox and label accessible via the keyboard. I've added a focus-visible style to the label whenever the checkbox is focused.

File: app/javascript/components/FetchingBoundary.tsx [hunk 2]
```diff
@@ -0,0 +1,45 @@
+import React from 'react'
```
Comment (kntsoriano) [inline app/javascript/components/FetchingBoundary.tsx:1]: This is a generic wrapper component that can be used when we're fetching data from an endpoint.

**Comment by iHiD** [comment]
@kntsoriano CSS tip for you: https://www.loom.com/share/3fb4158ebc274f0eb220eb3996d9377c

**Comment by iHiD** [comment]
@kntsoriano I've pushed a commit with a few bits:
1. Fixes the thing in loom
2. Refactors the CSS slightly so that the modifiers (`:checked+label`) have the minimal code in possible.
3. Fixed an issue where the modal would close if you clicked outside of the middle bit. This involved ensuring that the inner part of the modal covers 100% of the screen.
4. ~Changed `id` to `slug` everywhere. The JS wouldn't compile for me without this. I don't get how it worked at all. But maybe there's something I'm missing, or you missed pushing a file or something?~ I realised that you updated the serializer and added another commit to change it to `id` everywhere and set the `id` to be `slug` in the serializer.

**Comment by iHiD** [comment]
Rebased and force pushed so we can get the JS failures quicker.

**Comment by iHiD** [comment]
@kntsoriano The JS tests in `test/javascript/components/modals/mentor-registration-modal/ChooseTrackStep.test.js` this don't really work, both in terms of functionally being correct but also in terms of crashing the JS test runner in CI. I've tried to fix them but it seems to be quite wrong (e.g. the missing `selected`, `setSelected`, so I've commented it out and left it for you to fix.

File: app/javascript/utils/__mocks__/action-cable-consumer.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+  const subscription = {
+    unsubscribe: () => null,
+  }
+  return {
+    subscriptions: {
+      create: () => subscription,
+    },
+  }
+}
+export default createConsumer()
```
Comment (iHiD) [inline app/javascript/utils/__mocks__/action-cable-consumer.js:11]: @kntsoriano See this for how to do a mock everywhere properly.

File: test/javascript/setupTests.js [hunk 4]
```diff
@@ -3,6 +3,8 @@ import { waitFor, act } from '@testing-library/react'
 import { queryCache } from 'react-query'
 import flushPromises from 'flush-promises'
 
+jest.mock('../../app/javascript/utils/action-cable-consumer')
```
Comment (iHiD) [inline test/javascript/setupTests.js:6]: And then include this here.

This basically fixes everything.

Comment (iHiD) [reply test/javascript/setupTests.js:6]: (I hope!!!)

**Comment by iHiD** [comment]
@kntsoriano THE FUCKING JS TESTS ARE FUCKING GREEN MOTHERFUCKER.

**Pull request merged**